cmd/client-s3.go

```diff
@@ -168,7 +183,9 @@ func newFactory() func(config *Config) (Client, *probe.Error) {
 			// Cache the new minio client with hash of config as key.
 			clientCache[confSum] = api
 		}
-
+		if numParallelThreads := getNumMultiPartThreads(); numParallelThreads > 0 {
+			api.NumParallelWorkers = numParallelThreads
```

Comment:
I have another PR for minio/minio-go#794 where NumParallelWorkers is defined. There was a suggestion from @harshavardhana  on providing a Set method instead of exposing the api variable globally. So this code could change.